rotkehlchen/tasks/manager.py

```diff
@@ -976,6 +978,17 @@ def _maybe_query_graph_delegated_tokens(self) -> Optional[list[gevent.Greenlet]]
             addresses=self.chains_aggregator.accounts.eth,
         )]
 
+    def _maybe_recreate_historical_balances(self) -> Optional[list[gevent.Greenlet]]:
+        if should_run_periodic_task(self.database, DBCacheStatic.LAST_HISTORICAL_BALANCES_RECREATION_TS, HOUR_IN_SECONDS * 3) is False:  # noqa: E501
```

Comment:
This should not be a periodic task that runs every 3 hours. It's way too frequent. And if nothing has changed there is no reason to rerun.
Plus for exmple for rotki's account the way you wrote it right now it would take probably 4-6 weeks to run the task :rofl: 

This should be split into per asset tasks. And only if a new event for a specific asset has been added only balances **after** that event for that asset and for the total should be recalculated.

We already discussed this.